### PR TITLE
Display sub commands types only if there is any.

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Won't display "other commands" if there is no utility sub commands
- Won't display "other commands" if there is no sub commands

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: pkg/cmd/root.go [hunk 1]
```diff
@@ -83,3 +85,20 @@ func Root(p cli.Params) *cobra.Command {
...
+}
+func hasUtilitySubCommands(cmd *cobra.Command) bool {
+	return len(subCommands(cmd, "utility")) > 0
+}
+
+func subCommands(cmd *cobra.Command, annotation string) []*cobra.Command {
+	cmds := []*cobra.Command{}
+	for _, sub := range cmd.Commands() {
+		if sub.IsAvailableCommand() && sub.Annotations["commandType"] == annotation {
+			cmds = append(cmds, sub)
```
Comment (hrishin) [inline pkg/cmd/root.go]: can return `true` instead of populating any array, given the fact `cmds` are not used any where?

Comment (vdemeester) [reply pkg/cmd/root.go]: @hrishin maybe :angel: I was thinking we may want to use this later on and need to array but…

Comment (hrishin) [reply pkg/cmd/root.go]: Ahh was also wondering on the same if we can use it later.

Cool, nothing much to worry about it :smile:

**Pull request merged**